runtime/compiler/x/amd64/codegen/AMD64JNILinkage.cpp

```diff
@@ -521,40 +521,24 @@ void TR::AMD64JNILinkage::buildJNICallOutFrame(
    //    1:   frame flags
    //    0:   RAM method
    //
-   TR::X86RegImmInstruction *instr = generateRegImmInstruction(
-      SUBRegImms(),
-      callNode,
-      espReal,
-      _JNIDispatchInfo.numJNIFrameSlotsPushed * TR::Compiler->om.sizeofReferenceAddress(),
-      cg());
 
    // Build stack frame in Java stack.  Tag current bp.
    //
-   uintptrj_t tagBits = fej9->constJNICallOutFrameSpecialTag();
+   uintptrj_t tagBits = 0;
 
    // If the current method is simply a wrapper for the JNI call, hide the call-out stack frame.
    //
+   static_assert(IS_32BIT_SIGNED(J9SF_A0_INVISIBLE_TAG), "J9SF_A0_INVISIBLE_TAG must fit in immediate");
    if (resolvedMethod == comp()->getCurrentMethod())
-      tagBits |= fej9->constJNICallOutFrameInvisibleTag();
+      tagBits |= J9SF_A0_INVISIBLE_TAG;
 
    // Push tag bits (savedA0 slot).
    //
-   TR::MemoryReference *tempMR = generateX86MemoryReference(espReal, 4*TR::Compiler->om.sizeofReferenceAddress(), cg());
-   if (tagBits <= 0x7fffffff)
-      {
-      generateMemImmInstruction(SMemImm4(), callNode, tempMR, tagBits, cg());
-      }
-   else
-      {
-      if (!scratchReg)
-         scratchReg = cg()->allocateRegister();
-
-      generateRegImm64Instruction(MOV8RegImm64, callNode, scratchReg, tagBits, cg());
-      generateMemRegInstruction(S8MemReg, callNode, tempMR, scratchReg, cg());
-      }
+   generateImmInstruction(PUSHImm4, callNode, tagBits, cg());
 
    // (skip savedPC slot).
    //
+   generateImmInstruction(PUSHImm4, callNode, 0, cg());
```

Comment:
what is the `push 0` for?

Reply:
The linkage requires a slot for `savedPC`, but its content is ignored; therefore, I put this push instruction with an arbitrary value 0.